File: scripts/tr1/object_infos.lua
-- init defaults
local infos = {}
for i = 0, 190 do
    infos[i] = ObjectInfo.new()
end

-- Lara
infos[0].hit_points = 1000

-- Lara Twin
infos[6].hit_points = 1000

-- Wolf
infos[7].ai_agent = true
infos[7].radius = 341
infos[7].hit_points = 6
infos[7].pivot_length = 375
infos[7].target_update_chance = 0x2000

-- Bear
infos[8].ai_agent = true
infos[8].radius = 341
infos[8].hit_points = 20
infos[8].pivot_length = 500
infos[8].target_update_chance = 0x4000

-- Bat
infos[9].ai_agent = true
infos[9].radius = 102
infos[9].hit_points = 1
infos[9].target_update_chance = 0x400

-- Crocodile on land
infos[10].ai_agent = true
infos[10].radius = 341
infos[10].hit_points = 20
infos[10].pivot_length = 600
infos[10].target_update_chance = 0x2000

-- Crocodile in water
infos[11].ai_agent = true
infos[11].radius = 341
infos[11].hit_points = 20
infos[11].pivot_length = 600
infos[11].target_update_chance = 0x400

-- Lion (male)
infos[12].ai_agent = true
infos[12].radius = 341
infos[12].hit_points = 30
infos[12].pivot_length = 400
infos[12].target_update_chance = 0x7fff

-- Lion (female)
infos[13].ai_agent = true
infos[13].radius = 341
infos[13].hit_points = 25
infos[13].pivot_length = 400
infos[13].target_update_chance = 0x2000

-- Panther
infos[14].ai_agent = true
infos[14].radius = 341
infos[14].hit_points = 45
infos[14].pivot_length = 400
infos[14].target_update_chance = 0x2000

-- Gorilla
infos[15].ai_agent = true
infos[15].radius = 341
infos[15].hit_points = 22
infos[15].pivot_length = 250
infos[15].target_update_chance = 0x7fff

-- TRex
infos[18].ai_agent = true
infos[18].radius = 341
infos[18].hit_points = 100
infos[18].pivot_length = 2000
infos[18].target_update_chance = 0x7fff

-- Raptor
infos[19].ai_agent = true
infos[19].radius = 341
infos[19].hit_points = 20
infos[19].pivot_length = 400
infos[19].target_update_chance = 0x4000

-- Mummy
infos[24].hit_points = 18

-- Larson
infos[27].ai_agent = true
infos[27].radius = 102
infos[27].hit_points = 50
infos[27].target_update_chance = 0x7fff

-- Pierre
infos[28].ai_agent = true
infos[28].radius = 102
infos[28].hit_points = 70
infos[28].target_update_chance = 0x7fff

return infos

File: share/gameflows/jstrr/object_infos.py
from ce.engine import TR1ItemId, ObjectInfo
from ce.core import SectorSize

# init defaults
object_infos = {
    i: ObjectInfo()
    for i in TR1ItemId.__members__.values()
}

ob = object_infos[TR1ItemId.Lara]
ob.hit_points = 1000

ob = object_infos[TR1ItemId.Doppelganger]
ob.hit_points = 1000

ob = object_infos[TR1ItemId.Wolf]
ob.ai_agent = True
ob.radius = 341
ob.hit_points = 6
ob.pivot_length = 375
ob.target_update_chance = 0x2000
ob.drop_limit = -SectorSize

ob = object_infos[TR1ItemId.Bear]
ob.ai_agent = True
ob.radius = 341
ob.hit_points = 20
ob.pivot_length = 500
ob.target_update_chance = 0x4000

ob = object_infos[TR1ItemId.Bat]
ob.ai_agent = True
ob.radius = 102
ob.hit_points = 1
ob.target_update_chance = 0x400
ob.step_limit = 20 * SectorSize
ob.drop_limit = -20 * SectorSize
ob.fly_limit = 16

ob = object_infos[TR1ItemId.CrocodileOnLand]
ob.ai_agent = True
ob.radius = 341
ob.hit_points = 20
ob.pivot_length = 600
ob.target_update_chance = 0x2000

ob = object_infos[TR1ItemId.CrocodileInWater]
ob.ai_agent = True
ob.radius = 341
ob.hit_points = 20
ob.pivot_length = 600
ob.target_update_chance = 0x400
ob.step_limit = 20 * SectorSize
ob.drop_limit = -20 * SectorSize
ob.fly_limit = 16

ob = object_infos[TR1ItemId.LionMale]
ob.ai_agent = True
ob.radius = 341
ob.hit_points = 30
ob.pivot_length = 400
ob.target_update_chance = 0x7fff
ob.drop_limit = -SectorSize

ob = object_infos[TR1ItemId.LionFemale]
ob.ai_agent = True
ob.radius = 341
ob.hit_points = 25
ob.pivot_length = 400
ob.target_update_chance = 0x2000
ob.drop_limit = -SectorSize

ob = object_infos[TR1ItemId.Panther]
ob.ai_agent = True
ob.radius = 341
ob.hit_points = 45
ob.pivot_length = 400
ob.target_update_chance = 0x2000
ob.drop_limit = -SectorSize

ob = object_infos[TR1ItemId.Gorilla]
ob.ai_agent = True
ob.radius = 341
ob.hit_points = 22
ob.pivot_length = 250
ob.target_update_chance = 0x7fff
ob.step_limit = SectorSize // 2
ob.drop_limit = -SectorSize

ob = object_infos[TR1ItemId.RatOnLand]
ob.ai_agent = True
ob.radius = 204
ob.hit_points = 5
ob.pivot_length = 200
ob.target_update_chance = 0x2000

ob = object_infos[TR1ItemId.RatInWater]
ob.ai_agent = True
ob.radius = 204
ob.hit_points = 5
ob.pivot_length = 200
ob.target_update_chance = 0x2000

ob = object_infos[TR1ItemId.TRex]
ob.ai_agent = True
ob.radius = 341
ob.hit_points = 100
ob.pivot_length = 2000
ob.target_update_chance = 0x7fff
ob.cannot_visit_blockable = True
ob.cannot_visit_blocked = False

ob = object_infos[TR1ItemId.Raptor]
ob.ai_agent = True
ob.radius = 341
ob.hit_points = 20
ob.pivot_length = 400
ob.target_update_chance = 0x4000

ob = object_infos[TR1ItemId.FlyingMutant]
ob.ai_agent = True
ob.radius = 341
ob.hit_points = 50
ob.pivot_length = 150
ob.target_update_chance = 0x7fff
ob.cannot_visit_blockable = True
ob.cannot_visit_blocked = False

ob = object_infos[TR1ItemId.WalkingMutant1]
ob.ai_agent = True
ob.radius = 341
ob.hit_points = 50
ob.pivot_length = 150
ob.target_update_chance = 0x2000

ob = object_infos[TR1ItemId.WalkingMutant2]
ob.ai_agent = True
ob.radius = 341
ob.hit_points = 50
ob.pivot_length = 150
ob.target_update_chance = 0x2000

ob = object_infos[TR1ItemId.CentaurMutant]
ob.ai_agent = True
ob.radius = 341
ob.hit_points = 120
ob.pivot_length = 400
ob.target_update_chance = 0x7fff
ob.cannot_visit_blockable = True
ob.cannot_visit_blocked = False

ob = object_infos[TR1ItemId.Mummy]
ob.hit_points = 18

ob = object_infos[TR1ItemId.Larson]
ob.ai_agent = True
ob.radius = 102
ob.hit_points = 50
ob.target_update_chance = 0x7fff

ob = object_infos[TR1ItemId.Pierre]
ob.ai_agent = True
ob.radius = 102
ob.hit_points = 70
ob.target_update_chance = 0x7fff

ob = object_infos[TR1ItemId.SkateboardKid]
ob.ai_agent = True
ob.radius = 204
ob.hit_points = 125
ob.target_update_chance = 0x7fff

ob = object_infos[TR1ItemId.TorsoBoss]
ob.ai_agent = True
ob.radius = 341
ob.hit_points = 500
ob.target_update_chance = 0x7fff

ob = object_infos[TR1ItemId.Kold]
ob.ai_agent = True
ob.radius = 102
ob.hit_points = 200
ob.target_update_chance = 0x7fff

ob = object_infos[TR1ItemId.Cowboy]
ob.ai_agent = True
ob.radius = 102
ob.hit_points = 150
ob.target_update_chance = 0x7fff

ob = object_infos[TR1ItemId.Natla]
ob.ai_agent = True
ob.radius = 204
ob.hit_points = 400
ob.target_update_chance = 0x7fff

ob = object_infos[TR1ItemId.ScionPiece3]
ob.hit_points = 5

ob = object_infos[TR1ItemId.Fish]
ob.step_limit = 20 * SectorSize
ob.drop_limit = -20 * SectorSize
ob.fly_limit = 16

ob = object_infos[TR1ItemId.Sunglasses]
ob.hit_points = 1